### ISSUE TITLE
Analysing the use of the Application

### ISSUE THREAD

**Issue by markwilliamfirth** [dwyl/best-evidence#194]
As a product owner
I would like to be able to analyse the application
So that I can understand user behaviours and evaluate the MVP in order to give figures to funders to further its development

I require the following data and methods of analysis:
- [x] content of searches (linking these to individual users)
- [x] date of searches
- [x] publications clicked on from each search
- [x] notes made about search
- [x] notes made about question
- [x] Total number of registered users
- [x] Number of searches with notes
- [x] Number of searches with completed notes
- [x] Number of papers clicked through to with incomplete notes
- [x] Number of papers clicked through to with completed notes

**Averages to calculate derived from other metrics**
- [ ] Proportion of publications clicked to with incomplete notes = (No of incomplete publication notes / total no of publications clicked on)
- [ ] Proportion of publications clicked with completed notes = (No of complete publication notes / total no of publications clicked on)
- [ ] Average number of searches per user = ( total number of searches / number of users)

- [ ] Average number of publications clicked per search = (total number of publications clicked / number of searches)

**Time dependent stats**
- [ ] Number of searches made in last day/week/month/year/all time in total and by user
- [x] Proportion of registered users that have used app in last day/week/month/year/all time

**Total Flexibility**
- [ ] Free searching functionality (e.g. filter by different data such as date of search or single user ID)

These will either be achieved from:
https://github.com/dwyl/best-evidence/issues/146, or
Google Analytics, or
An alternative method

This issue is currently blocked by #146 - we first need to implement this in order to understand how many of the above requirements are solved by this solution.

@ajburls can you confirm in writing that all of the above are requirements for the MVP? Are any of the above not essential and therefore lower priority?

**Comment by ajburls** [comment]
Yes, confirmed

**Comment by Cleop** [comment]
I've been implementing the queries above on Redash. So that we comply with the privacy code for the app, we will set @ajburls up with a Redash account that is read-only. This way there is no danger that @ajburls can access people's email addresses and also that you can inadvertently alter the live db. 

The drawback to this implementation is that it means that @ajburls cannot write her own queries, however I have been creating queries for you. You can tell what data that you requested above I have yet covered by those points that do not have a ticked checkbox. The averages section should be possible for you to calculate manually although if I have the time I will write custom queries for this. You can download the data from any of the queries into excel and analyse them there too. 

@ajburls - I have sent you an invite to use Redash. Once you have done this I will amend your settings so that you have read-only.

**Comment by ajburls** [comment]
@cleop I can run the queries you have created and these are great. When we spoke at the stand up I thought we agreed you would give me full access rights so I could create new queries (despite the fact that I could if I wanted to see people's IDs). I have tried creating new queries but can't so I think that maybe you didn't do this. Please let me know.

**Comment by Cleop** [comment]
@ajburls - you are in the default group which I have set to 'full access' under the permissions. You should have access to everything, I just checked it was the case. I presume you're trying to make a new query by going to:
<img src="https://user-images.githubusercontent.com/16775804/30640669-b533d5be-9dfb-11e7-9583-af78d98237aa.png" width=200px />
It should then show this page:
![image](https://user-images.githubusercontent.com/16775804/30640678-be013182-9dfb-11e7-9153-b1d7c8ec0aef.png)

**Comment by ajburls** [comment]
I have now got it working.  Well not exactly working properly but I have been able to start playing.
